apps/auracast.py

```diff
@@ -274,12 +274,24 @@ async def stop(self) -> None:
         await self.device.stop_scanning()
 
     def on_advertisement(self, advertisement: bumble.device.Advertisement) -> None:
-        if (
-            broadcast_name := advertisement.data.get(
-                bumble.core.AdvertisingData.BROADCAST_NAME
+        if not (
+            ads := advertisement.data.get_all(
+                bumble.core.AdvertisingData.SERVICE_DATA_16_BIT_UUID
             )
-        ) is None:
+        ) or not (
+            any(
+                ad
+                for ad in ads
+                if isinstance(ad, tuple)
+                and ad[0] == bumble.gatt.GATT_BROADCAST_AUDIO_ANNOUNCEMENT_SERVICE
+            )
+        ):
             return
+
+        broadcast_name = (
+            advertisement.data.get(bumble.core.AdvertisingData.BROADCAST_NAME)
+            or 'Unknown'
```

Comment:
there's a (small) risk of conflict here. If a real broadcast service uses the name 'Unknown', we don't know the difference. Maybe the best would be to allow passing `None` for the `broadcast_name` parameter of `self.on_new_broadcast`?

Reply:
Done. Skip name when it's None here.